server/route.go

```diff
@@ -62,12 +61,196 @@ type connectInfo struct {
 	Name     string `json:"name"`
 }
 
+// Used to hold onto mappings for unsubscribed
+// routed queue subscribers.
+type rqsub struct {
+	group []byte
+	atime time.Time
+}
+
 // Route protocol constants
 const (
 	ConProto  = "CONNECT %s" + _CRLF_
 	InfoProto = "INFO %s" + _CRLF_
 )
 
+// Clear up the timer and any map held for remote qsubs.
+func (s *Server) clearRemoteQSubs() {
+	s.rqsMu.Lock()
+	defer s.rqsMu.Unlock()
+	if s.rqsubsTimer != nil {
+		s.rqsubsTimer.Stop()
+		s.rqsubsTimer = nil
+	}
+	s.rqsubs = nil
+}
+
+// Check to see if we can remove any of the remote qsubs mappings
+func (s *Server) purgeRemoteQSubs() {
+	ri := s.getOpts().RQSubsSweep
+	s.rqsMu.Lock()
+	exp := time.Now().Add(-ri)
+	for k, rqsub := range s.rqsubs {
+		if exp.After(rqsub.atime) {
+			delete(s.rqsubs, k)
+		}
+	}
+	if s.rqsubsTimer != nil {
+		// Reset timer.
+		s.rqsubsTimer = time.AfterFunc(ri, s.purgeRemoteQSubs)
+	}
+	s.rqsMu.Unlock()
+}
+
+// Lookup a remote queue group sid.
+func (s *Server) lookupRemoteQGroup(sid string) []byte {
+	s.rqsMu.RLock()
+	rqsub := s.rqsubs[sid]
+	s.rqsMu.RUnlock()
+	return rqsub.group
```

Comment:
In theory, the group could not be found...

Reply:
That is handled via a DebugF on route.go:159

Reply:
Ok, I see.